realm/realm-library/src/objectServer/java/io/realm/mongodb/sync/Sync.java

```diff
@@ -307,7 +323,7 @@ synchronized void reset() {
      * @param session Session to trigger Client Reset for.
      */
     void simulateClientReset(SyncSession session) {
-        simulateClientReset(session, ErrorCode.DIVERGING_HISTORIES);
+        new Thread(() -> simulateClientReset(session, ErrorCode.DIVERGING_HISTORIES)).start();
```

Comment:
Maybe add a comment on why we run it on a separate thread (to simulate the error coming from the sync writer thread)